pyfstat/make_sfts.py

```diff
@@ -441,50 +445,61 @@ def check_cached_data_okay_to_use(self, cl_mfd):
+        # matching CLs
+        for sftfile in sftfiles:
+            catalog = lalpulsar.SFTdataFind(sftfile, None)
+            cl_old = helper_functions.get_lalapps_commandline_from_SFTDescriptor(
+                catalog.data[0]
             )
-            return False
-        if not helper_functions.match_commandlines(cl_old, cl_mfd):
-            logging.info("Commandlines unmatched. " + need_new)
-            return False
-        else:
-            logging.info("OK: Commandline matched with old SFT header.")
-        logging.info("Looks like cached data matches current options, will re-use it!")
+            if len(cl_old) == 0:
+                logging.info(
+                    "......ould not obtain comparison commandline from first SFT"
+                    " header in old file '{}'. {}".format(sftfile, need_new)
+                )
+                return False
+            if not helper_functions.match_commandlines(cl_old, cl_mfd):
+                logging.info(
+                    "......commandlines unmatched for first SFT in old"
+                    " file '{}'. {}".format(sftfile, need_new)
+                )
+                return False
+        logging.info("......OK: Commandline matched with old SFT header(s).")
+        logging.info(
+            "...all data consistency checks passed: Looks like existing"
+            " SFT data matches current options, will re-use it!"
+        )
         return True
 
-    def check_if_cff_file_needs_rewritting(self, content):
+    def check_if_cff_file_needs_rewriting(self, content):
         """Check if the .cff file has changed
 
         Returns True if the file should be overwritten - where possible avoid
         overwriting to allow cached data to be used
         """
         logging.info("Checking if we can re-use injection config file...")
         if os.path.isfile(self.config_file_name) is False:
-            logging.info("No config file {} found.".format(self.config_file_name))
+            logging.info("...no config file {} found.".format(self.config_file_name))
             return True
         else:
-            logging.info("Config file {} already exists.".format(self.config_file_name))
+            logging.info(
+                "...OK: config file {} already exists.".format(self.config_file_name)
+            )
 
         with open(self.config_file_name, "r") as f:
             file_content = f.read()
             if file_content == content:
                 logging.info(
-                    "File contents match, no update of {} required.".format(
+                    "...OK: file contents match, no update of {} required.".format(
                         self.config_file_name
                     )
                 )
                 return False
             else:
                 logging.info(
-                    "File contents unmatched, updating {}.".format(
+                    "...file contents unmatched, updating {}.".format(
```

Comment:
Just a quick question:

This way of comparing two files assumes two differently formatted files to be different even when they contain the very same signal parameters.

Hence, by doing things in this way we are stating "If you want to use PyFstat in all its glory, please follow the right and obvious way to do an injection and stop messing with `.cff` files yourself.".

I'm Ok with this. Any comments/objections?

Reply:
We could use set operations as in `helper_functions.match_commandlines()` over the set of lines to allow for re-ordering, and whitespace-ignoring comparisons on each line. Let's make a reminder issue for that as a future improvement.

Reply:
#165